modules/pipeline/providers/leaderworker/etcd.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) 2021 Terminus, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package leaderworker
+
+import (
+	"context"
+	"path/filepath"
+
+	"github.com/coreos/etcd/clientv3"
+
+	"github.com/erda-project/erda/modules/pipeline/providers/leaderworker/worker"
+	"github.com/erda-project/erda/pkg/strutil"
+)
+
+func (p *provider) notifyWorkerAdd(ctx context.Context, w worker.Worker, typ worker.Type) error {
```

Comment:
is `addWorkerAsync` better? or `asyncAddWorker`

Reply:
no

Reply:
maybe `notify` makes you confused.
I changed to `registerWorker`, because this func do register worker to leader-worker framework, report heartbeat and then put key into etcd.